casanovo/denovo/model.py

```diff
@@ -759,19 +764,26 @@ def validation_step(
         for spectrum_preds in self.forward(batch[0], batch[1]):
             for _, _, pred in spectrum_preds:
                 peptides_pred.append(pred)
-        aa_precision, _, pep_precision = evaluate.aa_match_metrics(
-            *evaluate.aa_match_batch(
-                peptides_pred, peptides_true, self.decoder._peptide_mass.masses
-            )
-        )
-        log_args = dict(on_step=False, on_epoch=True, sync_dist=True)
-        self.log(
-            "Peptide precision at coverage=1",
-            pep_precision,
-            **log_args,
-        )
-        self.log("AA precision at coverage=1", aa_precision, **log_args)
 
+        if self.calculate_precision:
```

Comment:
**question:** This only doesn't calculate the metrics, but peptide inference is already done in the preceding lines I think? The latter is the very expensive step, the former adds only a small amount of extra runtime.

Reply:
You're right - it should be fixed!